### PR TITLE
Check file extension when running/debugging file #1037

### LINKED ISSUE #1037: Ctrl + F5 Request
When doing assembly invoking it should prompt us some warning message if file extension is not .exe
for example i accidentally executed malware even when it was renamed as .bit because i was working in parralel with visual studio and i clicked ctrl + f5 on dnspy and than it executed malware on my pc

### PR DESCRIPTION
**Pull request by SilentHammerHUN**
Closes #1037

### CODE REVIEW

File: Extensions/dnSpy.Debugger/dnSpy.Debugger/DbgUI/DebuggerImpl.cs [hunk 1]
```diff
@@ -100,6 +100,14 @@ sealed class DebuggerImpl : Debugger, IDbgManagerStartListener {
 		public override void StartWithoutDebugging() {
 			if (!CanStartWithoutDebugging)
 				return;
+			var fileExtension = GetCurrentExecutableFilename();
```
Comment (0xd4d) [inline Extensions/dnSpy.Debugger/dnSpy.Debugger/DbgUI/DebuggerImpl.cs]: This method can return null so you need a null check.

File: Extensions/dnSpy.Debugger/dnSpy.Debugger/DbgUI/DebuggerImpl.cs [hunk 2]
```diff
@@ -100,6 +100,14 @@ sealed class DebuggerImpl : Debugger, IDbgManagerStartListener {
...
+			fileExtension = fileExtension.Substring(fileExtension.Length - 4);
```
Comment (0xd4d) [inline Extensions/dnSpy.Debugger/dnSpy.Debugger/DbgUI/DebuggerImpl.cs]: Path.GetExtension() + StringComparer.OrdinalIgnoreCase.Equals()

File: Extensions/dnSpy.Debugger/dnSpy.Debugger/DbgUI/DebuggerImpl.cs [hunk 3]
```diff
@@ -100,6 +100,14 @@ sealed class DebuggerImpl : Debugger, IDbgManagerStartListener {
...
+			if (fileExtension != ".exe") {
+				var result = messageBoxService.Value.Show(dnSpy_Debugger_Resources.RunFileWithNonExeExtension, MsgBoxButton.Yes | MsgBoxButton.No);
+				if (result == MsgBoxButton.None || result == MsgBoxButton.No) {
```
Comment (0xd4d) [inline Extensions/dnSpy.Debugger/dnSpy.Debugger/DbgUI/DebuggerImpl.cs]: Curly braces aren't needed if it's just one statement in the block.

File: Extensions/dnSpy.Debugger/dnSpy.Debugger/DbgUI/DebuggerImpl.cs [hunk 4]
```diff
@@ -108,6 +116,14 @@ sealed class DebuggerImpl : Debugger, IDbgManagerStartListener {
 		public override void DebugProgram(bool pauseAtEntryPoint) {
 			if (!CanDebugProgram)
 				return;
+			var fileExtension = GetCurrentExecutableFilename();
```
Comment (0xd4d) [inline Extensions/dnSpy.Debugger/dnSpy.Debugger/DbgUI/DebuggerImpl.cs]: Same issues as I wrote above

File: Extensions/dnSpy.Debugger/dnSpy.Debugger/Properties/dnSpy.Debugger.Resources.hu.resx [hunk 5]
```diff
@@ -59,46 +59,46 @@
             : using a System.ComponentModel.TypeConverter
             : and then encoded with base64 encoding.
     -->
-  <xsd:schema xmlns="" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:msdata="urn:schemas-microsoft-com:xml-msdata" id="root">
```
Comment (0xd4d) [inline Extensions/dnSpy.Debugger/dnSpy.Debugger/Properties/dnSpy.Debugger.Resources.hu.resx]: This file gets auto updated when I download the latest translations from crowdin, so please restore it.

**Comment by 0xd4d** [comment]
eg. something like
```c#
var extension = Path.GetExtension(GetCurrentExecutableFilename());
if (!StringComparer.Ordinal.Equals(extension, ".exe")) ...
```

**Comment by 0xd4d** [comment]
I added `Closes #1037` to your 1st post so the issue gets closed when it gets merged

**Comment by SilentHammerHUN** [comment]
That should do it

**Comment by 0xd4d** [comment]
I just realized that .NET Core files have a dll extension so this will unfortunately not work. I probably need to add a new API to get some more info that can be used by this code.

**Comment by 0xd4d** [comment]
I've pushed a commit that has new code you can use to check for the wrong extension (in the same file you edited)

```
if (!startDebuggingOptionsProvider.Value.CanStartWithoutDebugging(out var result))
```

and

```
var (options, flags) = startDebuggingOptionsProvider.Value.GetStartDebuggingOptions(breakKind);
```

`result` and `flags` can be checked. If the `WrongExtension` bit is set, you can show the message.

Since this code doesn't know which extension is valid, the message shouldn't contain '.exe', but should be more general eg. "invalid extension" or something.

**Comment by SilentHammerHUN** [comment]
@0xd4d that should be it

**Comment by 0xd4d** [comment]
You need to rebase this code on the current master. You manually added the code, but that's the wrong way and the code is impossible to merge.

Try pulling in latest changes, merge, and rebase. You can also create a new branch based on master and manually add your code there and push it here.

**Comment by 0xd4d** [comment]
The easiest fix is probably just to get latest master and then add your changes to it then force push it here again.

**Comment by SilentHammerHUN** [comment]
@0xd4d Done

**Comment by 0xd4d** [comment]
Thanks, it's been merged now!

**Pull request merged**